javaee/api/src/main/java/org/jboss/forge/addon/javaee/servlet/SecurityConstraintDescriptor.java

```diff
@@ -0,0 +1,177 @@
+package org.jboss.forge.addon.javaee.servlet;
+
+import org.jboss.forge.addon.javaee.security.TransportGuarantee;
+
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * Descriptor of the security constraint element part of the web.xml
+ *
+ * @author <a href="mailto:ivan.st.ivanov@gmail.com">Ivan St. Ivanov</a>
+ */
+public class SecurityConstraintDescriptor
```

Comment:
This rework will also benefit from having a single descriptor to work on. So there is no need to copy/paste the code on each ServletFacet impl<div>
</div><div>
</div><div>-------- Mensagem original --------</div><div>De: "Ivan St. Ivanov" <notifications@github.com> </div><div>Data: 20/07/2015  17:45  (GMT-03:00) </div><div>Para: forge/core <core@noreply.github.com> </div><div>Cc: George Gastaldi <ggastald@redhat.com> </div><div>Assunto: Re: [core] FORGEPLUGINS-152: Add some JavaEE security commands (#570) </div><div>
</div>> @@ -0,0 +1,177 @@
> +package org.jboss.forge.addon.javaee.servlet;
> +
> +import org.jboss.forge.addon.javaee.security.TransportGuarantee;
> +
> +import java.util.ArrayList;
> +import java.util.List;
> +
> +/**
> + * Descriptor of the security constraint element part of the web.xml
> + *
> + * @author <a href="mailto:ivan.st.ivanov@gmail.com">Ivan St. Ivanov</a>
> + */
> +public class SecurityConstraintDescriptor

We will need to rework a little bit the descriptors part anyway. Two reasons besides the one discussed above:
* The web app descriptor does not support removing a single role or constraint (or anything else). It supports removing all of them
* At the moment the SecurityConstraintType returns a list of WebResourceCollections. However, according to the spec, one security constraint element holds just one web resource collection

---
Reply to this email directly or view it on GitHub:
https://github.com/forge/core/pull/570/files#r35041842